packages/batch-submitter/src/transaction-chain-contract.ts

```diff
@@ -19,6 +20,29 @@ export { encodeAppendSequencerBatch, BatchContext, AppendSequencerBatchParams }
  * where the `appendSequencerBatch(...)` function uses a specialized encoding for improved efficiency.
  */
 export class CanonicalTransactionChainContract extends Contract {
+  public customPopulateTransaction = {
+    appendSequencerBatch: async (
+      batch: AppendSequencerBatchParams,
+      options?: TransactionRequest
+    ): Promise<ethers.PopulatedTransaction> => {
+      const nonce = await this.signer.getTransactionCount()
+      const to = this.address
+      const data = getEncodedCalldata(batch)
+      const gasLimit = await this.signer.provider.estimateGas({
+        to,
+        from: await this.signer.getAddress(),
+        data,
+      })
+      const value = 0
```

Comment:
`value` looks unused?

Reply:
Yeah I just set it to zero explicitly. Probably could leave it out entirely but was just making zero value explicit. Let me know if you want me to remove that line

Reply:
Its not returned?

Reply:
Ah you're right -- deleted it